src/DynamoCore/Engine/MigrateFormulaToDS.cs

```diff
@@ -0,0 +1,101 @@
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text.RegularExpressions;
+using ProtoCore;
+using ProtoCore.AST.AssociativeAST;
+using ProtoCore.SyntaxAnalysis;
+using ProtoCore.Utils;
+
+namespace Dynamo.Engine
+{
+    internal class MigrateFormulaToDS : AstReplacer
+    {
+        private static readonly TextInfo textInfo = CultureInfo.InvariantCulture.TextInfo;
+
+        /// <summary>
+        /// Regex to match "if(<condition>, <true>, <false>)" pattern in Formula node and extract the parts of the
+        /// "if" statement to construct the corresponding conditional AST in DS. This is because the DS parser does not
+        /// recognize this "if" syntax.
+        /// </summary>
+        private const string ifCond = @"(if\s*)\(\s*(.*?)\s*\)$";
+        private static readonly Regex ifRgx = new Regex(ifCond, RegexOptions.IgnoreCase);
+
+
+        internal string ConvertFormulaToDS(string formula)
+        {
+            CodeBlockNode cbn = null;
+            var match = ifRgx.Match(formula);
+
+            List<AssociativeNode> dsAst = new List<AssociativeNode>();
+            if (match.Success)
+            {
+                var cond = match.Groups[2].Value;
```

Comment:
I am not Familiar with RegEx APIs. Are we always going to have match.Groups.Size > 2 ? And is it always at that position ?

Reply:
thanks for clarifying